Source/WebCore/platform/graphics/ImageBuffer.cpp

```diff
@@ -227,13 +227,28 @@ RefPtr<NativeImage> ImageBuffer::copyNativeImageForDrawing(BackingStoreCopy copy
 
 RefPtr<NativeImage> ImageBuffer::sinkIntoNativeImage()
 {
+
     if (auto* backend = ensureBackendCreated()) {
         flushDrawingContext();
         return backend->sinkIntoNativeImage();
     }
     return nullptr;
 }
 
+RefPtr<ImageBuffer> ImageBuffer::sinkIntoBufferForDifferentThread(RefPtr<ImageBuffer> buffer)
+{
+    if (!buffer)
+        return nullptr;
+    ASSERT(buffer->hasOneRef());
+    return buffer->sinkIntoBufferForDifferentThread();
+}
+
+RefPtr<ImageBuffer> ImageBuffer::sinkIntoBufferForDifferentThread()
+{
+    ASSERT(hasOneRef());
```

Comment:
As above, the naming might not be great here.

It asserts that there's only one ref, because it's a private virtual and the only caller is the static which also asserts that it owns the only ref.

The default behaviour is that if the static owns the only ref to the ImageBuffer, then we can allow it to be passed to a new thread as-is since there won't be any other cross-thread references.